From 5f406a2627f0d4ab6b95e02d469fdeabc85bea09 Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Tue, 8 Mar 2022 22:06:07 +0800
Subject: [PATCH] efi: Fix efi_find_mirror redefine in x86
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5f406a2627f0d4ab6b95e02d469fdeabc85bea09
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SK3S
CVE: NA

----------------------------------------------

Commit cc3d801fa10f ("efi: Make efi_find_mirror() public") add
efi_find_mirror() defination into linux/efi.h, but forget to
drop this in arch/x86/include/asm/efi.h, kill it.

Fixes: cc3d801fa10f ("efi: Make efi_find_mirror() public")
Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/x86/include/asm/efi.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/x86/include/asm/efi.h b/arch/x86/include/asm/efi.h
index 3be8754408d5..3f58bc3fb550 100644
--- a/arch/x86/include/asm/efi.h
+++ b/arch/x86/include/asm/efi.h
@@ -353,7 +353,6 @@ static inline bool efi_is_64bit(void)
 extern bool efi_reboot_required(void);
 extern bool efi_is_table_address(unsigned long phys_addr);
 
-extern void efi_find_mirror(void);
 extern void efi_reserve_boot_services(void);
 #else
 static inline void parse_efi_setup(u64 phys_addr, u32 data_len) {}
@@ -365,9 +364,6 @@ static inline  bool efi_is_table_address(unsigned long phys_addr)
 {
 	return false;
 }
-static inline void efi_find_mirror(void)
-{
-}
 static inline void efi_reserve_boot_services(void)
 {
 }
-- 
2.26.2

